lib/aiken/trees/mt.ak

```diff
@@ -0,0 +1,504 @@
+use acca/either.{Either, Left, Right}
+use aiken/bytearray.{from_string}
+use aiken/hash.{Hash, Sha2_256, sha2_256}
+use aiken/list
+use aiken/option.{is_none}
+
+// Construction of the merkle tree shouldn't be done by hand, but via
+// 'from_list'.
+/// A purely functional implementation of MerkleTrees that is suitable for
+/// usage on-chain. Note however that the construction of 'MerkleTree' and
+/// membership proofs are still expected to happen *off-chain* while only the
+/// proof verification should be done on-chain.
+///
+/// Code ported to Aiken from Hydra (https://github.com/input-output-hk/hydra/blob/master/plutus-merkle-tree/src/Plutus/MerkleTree.hs)
+/// A MerkleTree representation, suitable for on-chain manipulation.
+pub type MerkleTree {
+  Empty
+  Leaf { hash: Hash<Sha2_256, ByteArray> }
+  Node { hash: Hash<Sha2_256, ByteArray>, left: MerkleTree, right: MerkleTree }
+}
+
+pub type Proof =
+  List<Either<Hash<Sha2_256, ByteArray>, Hash<Sha2_256, ByteArray>>>
+
+fn do_to_list(node: MerkleTree) -> List<Hash<Sha2_256, ByteArray>> {
```

Comment:
This function adds no benefits because it's called directly from `to_list`. Please inline.